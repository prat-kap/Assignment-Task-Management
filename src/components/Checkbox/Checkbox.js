import { Form } from "react-bootstrap"
import PropTypes from "prop-types"

const Checkbox = props => {
  const { label, id, onClick, checkboxRef, disabled } = props
  return (
    <Form.Check
      type={"checkbox"}
      id={id}
      label={label}
      onClick={onClick}
      ref={checkboxRef}
      disabled={disabled}
      data-testid="default-checkbox"
    />
  )
}

Checkbox.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  checkboxRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any })
  ])
}
export default Checkbox
