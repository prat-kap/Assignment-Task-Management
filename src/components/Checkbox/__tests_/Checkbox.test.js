import { BrowserRouter as Router } from "react-router-dom"
import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

import Checkbox from "../Checkbox"
import Loading from "../../Loader/Loading"
import "../../../i18n"

it("Checkbox is checked/unchecked", () => {
  render(
    <Router>
      <Checkbox />
    </Router>
  )
  const user = userEvent.setup()
  const checkbox = user.type(screen.getByTestId("default-checkbox"))
  expect(checkbox.checked).toEqual()
  user.click(checkbox)
  const { container } = render(<Loading />)
  expect(container).toMatchSnapshot()
  user.click(checkbox)
  expect(container).toMatchSnapshot()
  expect(checkbox.checked).toEqual()
})
