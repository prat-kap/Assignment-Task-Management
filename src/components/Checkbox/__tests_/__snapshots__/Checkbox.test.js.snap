// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox is checked/unchecked 1`] = `
<div>
  <div
    style="display: flex; justify-content: center; align-items: center; width: 100%; height: 100%;"
  >
    <div
      class="spinner-border"
      style="width: 80px; height: 80px;"
    />
  </div>
</div>
`;

exports[`Checkbox is checked/unchecked 2`] = `
<div>
  <div
    style="display: flex; justify-content: center; align-items: center; width: 100%; height: 100%;"
  >
    <div
      class="spinner-border"
      style="width: 80px; height: 80px;"
    />
  </div>
</div>
`;
